Sources/OpenAISwift/Models/ChatMessage.swift

```diff
@@ -34,6 +35,26 @@ public struct ChatMessage: Codable, Identifiable {
         self.role = role
         self.content = content
     }
+
+    enum CodingKeys: CodingKey {
+        case role
+        case content
+    }
+
+    public init(from decoder: Decoder) throws {
```

Comment:
@marcoboerner I may be missing something, but I don't think this is needed. Specifying `CodingKeys` should be sufficient. I have a fork with a commit [here](https://github.com/aronbudinszky/OpenAISwift/commit/23bad31a49b49cc5b6d4965bceaed6c664774bd3) that fixes this issue.

(Same goes for `encode()` method below.)

For what it's worth I'll open up a PR with the "one line" solution that _only_ solves the reported problem (maybe that'll get merged sooner?).

That said the other changes you made are also useful...